### PR TITLE
Adding abstract overrides in RuleFilter

### PR DESCRIPTION
**Pull request by ki1729**
# Description

This change is to isolate changes to the abstract RuleFilter class. 
Formalize implementation of euqals and hashcode methods by all RuleFilter implementations

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by JonathanGiles** [comment]
I'm not inclined to support this, just because it is a breaking change for negligible gain. I would prefer if we could document this as being a requirement, which seems reasonable to me as it is something developers should be expected to provide anyway, and documentation is often how this is enforced in my experience.

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/models/RuleFilter.java [hunk 1]
```diff
@@ -27,4 +27,11 @@
         // This is intentionally left blank. This constructor exists
         // only to prevent external assemblies inheriting from it.
     }
+
+    @Override
+    public abstract boolean equals(Object obj);
+
+    @Override
+    public abstract int hashCode();
```
Comment (srnagar) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/models/RuleFilter.java:35]: As discussed offline, this is a breaking change. If customers have already extended from this abstract class without overriding `equals` and `hashCode` methods, then introducing these abstract methods here will break them.

**Comment by ki1729** [comment]
Closing this out - this is something to be added when there is a major version bump

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/models/RuleFilter.java (hunk 1 above)
Comment (ki1729) [reply sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/models/RuleFilter.java:35]: Agreed, makes sense closed out this PR - Let's add this to the To-Do list for major version bump

**Pull request closed without merging**